src/reporting/Market.sol

```diff
@@ -0,0 +1,653 @@
+        require((DISPLAY_PRICE_MIN <= _maxDisplayPrice) && (_maxDisplayPrice <= DISPLAY_PRICE_MAX));
+        require((DISPLAY_PRICE_MIN <= _minDisplayPrice) && (_minDisplayPrice <= DISPLAY_PRICE_MAX));
+        require((1 <= uint256(_maxDisplayPrice.sub(_minDisplayPrice))) && (uint256(_maxDisplayPrice.sub(_minDisplayPrice)) <= SET_COST_MULTIPLIER_MAX));
+        require(_creator != NULL_ADDRESS);
+        // FIXME: require market to be on a non-forking branch; repeat this check up the stack as well if necessary (e.g., in reporting window)
+        // CONSIDER: should we allow creator to send extra ETH, is there risk of variability in bond requirements?
+        require(msg.value == MarketFeeCalculator(controller.lookup("MarketFeeCalculator")).getMarketCreationCost(_reportingWindow));
+        reportingWindow = _reportingWindow;
+        endTime = _endTime;
+        numOutcomes = _numOutcomes;
+        payoutDenominator = _payoutDenominator;
+        // FIXME: markets may be denominated in tokens that aren't 10^18, deal with that
+        feePerEthInAttoeth = _feePerEthInAttoeth;
+        maxDisplayPrice = _maxDisplayPrice;
+        minDisplayPrice = _minDisplayPrice;
+        marketCreationBlock = block.number;
+        topic = _topic;
+        automatedReporterAddress = _automatedReporterAddress;
+        denominationToken = _denominationToken;
+        creator = _creator;
+        for (uint8 _outcome = 0; _outcome < numOutcomes; _outcome++) {
+            shareTokens.push(createShareToken(_outcome));
+        }
+        approveSpenders();
+        require(controller.lookup("Cash") == address(denominationToken) || getBranch().isContainerForShareToken(denominationToken));
+        _success = true;
+        return _success;
+    }
+
+    function createShareToken(uint8 _outcome) private returns (IShareToken) {
+        return ShareTokenFactory(controller.lookup("ShareTokenFactory")).createShareToken(controller, this, _outcome);
+    }
+
+        // TODO: we need to update this signature (and all of the places that call it) to allow the creator (UI) to pass in a number of other things which will all be logged here
+        // TODO: log short description
+        // TODO: log long description
+        // TODO: log min display price
+        // TODO: log max display price
+        // TODO: log tags (0-2)
+        // TODO: log outcome labels (same number as numOutcomes)
+        // TODO: log type (scalar, binary, categorical)
+        // TODO: log any immutable data associated with the market (e.g., endTime, numOutcomes, payoutDenominator, denominationToken address, etc.)
+
+    // this will need to be called manually for each open market if a spender contract is updated
+    function approveSpenders() private returns (bool) {
+        bytes32[5] memory _names = [bytes32("cancelOrder"), bytes32("completeSets"), bytes32("takeOrder"), bytes32("tradingEscapeHatch"), bytes32("claimProceeds")];
+        uint8 i = 0;
```

Comment:
Style: I'm guessing you did this because Solidity cries about re-using the variable name, despite the scope difference (if that is the case I encourage you to submit a bug to Solidity since `for(uint8 i = 0; i < x; i++) { }` should scope `i` to inside of the for loop)?  Assuming that is the case, I think I would personally prefer to just have one loop use `i` and the other use `j` than declare it up here.  Doing it this way gave me pause, but `i` and `j` wouldn't have.